pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/TrigonometricTransformFunctions.java

```diff
@@ -0,0 +1,281 @@
+
+import com.google.common.base.Preconditions;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+
+
+public class TrigonometricTransformFunctions {
+  public static class Atan2TransformFunction extends BaseTransformFunction {
+    public static final String FUNCTION_NAME = "atan2";
+    private double[] _result;
+    private TransformFunction _leftTransformFunction;
+    private TransformFunction _rightTransformFunction;
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+      // Check that there are more than 1 arguments
+      if (arguments.size() != 2) {
+        throw new IllegalArgumentException("Exactly 2 arguments are required for Atan2 transform function");
+      }
+
+      _leftTransformFunction = arguments.get(0);
+      _rightTransformFunction = arguments.get(1);
+      Preconditions.checkArgument(
+          _leftTransformFunction.getResultMetadata().isSingleValue() || _rightTransformFunction.getResultMetadata()
+              .isSingleValue(), "Argument must be single-valued for transform function: %s", getName());
+    }
+
+    @Override
+    public TransformResultMetadata getResultMetadata() {
+      return DOUBLE_SV_NO_DICTIONARY_METADATA;
+    }
+
+    @Override
+    public double[] transformToDoubleValuesSV(ProjectionBlock projectionBlock) {
+      int length = projectionBlock.getNumDocs();
+
+      if (_result == null || _result.length < length) {
+        _result = new double[length];
+      }
+
+      double[] leftValues = _leftTransformFunction.transformToDoubleValuesSV(projectionBlock);
+      double[] rightValues = _rightTransformFunction.transformToDoubleValuesSV(projectionBlock);
+      for (int i = 0; i < length; i++) {
+        _result[i] = Math.atan2(leftValues[i], rightValues[i]);
+      }
+
+      return _result;
+    }
+  }
+
+  public static class DegreesTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "degrees";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected void applyMathOperator(double[] values, int length) {
+      for (int i = 0; i < length; i++) {
+        _results[i] = Math.toDegrees(values[i]);
+      }
+      ;
+    }
+  }
+
+  public static class AcosTransformFunction extends SingleParamMathTransformFunction {
+    public static final String FUNCTION_NAME = "acos";
+
+    @Override
+    public String getName() {
+      return FUNCTION_NAME;
+    }
+
+    @Override
+    protected void applyMathOperator(double[] values, int length) {
+      for (int i = 0; i < length; i++) {
+        _results[i] = Math.acos(values[i]);
+      }
+      ;
```

Comment:
Can we just get rid of these semicolons please?

Reply:
My bad. Missed in this file. Removed now.